### DISCUSSION TITLE
Get cost and usage for ec2 instances

### DISCUSSION THREAD

**Discussion by arunppsg** [aws/aws-cli#8229] [Q&A]
I want to find usage cost of `t3.medium` instance and I used the following command:
```
aws ce get-cost-and-usage --time-period Start=2023-10-01,End=2023-10-05 \
--granularity=DAILY --metrics "UnblendedCost" \
--group-by Type=DIMENSION,Key=INSTANCE_TYPE \
 --filter '{ "Dimensions": { "Key": "INSTANCE_TYPE", "MatchOptions": ["EQUALS"], "Values": ["t3.large"] }}' 
```

but the output shows total unblended costs as 0
```
"Total": {
      "UnblendedCost": {
          "Amount": "0",
          "Unit": "USD"
      }
```

though t3.large instance has been running during this period. Am I doing something wrong here? How can I get usage cost of t3.large instance during this period?

**Comment by RyanFitzSimmonsAK** [comment]
To clarify, are you trying to get the usage cost of a `t3.large` instance, or a `t3.medium` instance?  Your code says medium, but you said that you've been running a `t3.large` instance.

**Reply by arunppsg** [reply]
It's `t3.large` - apologies for the confusion, I was using both these kind of instances and testing it out for various instance types. In both cases, the results show total unblended cost as 0. Also, updated the code.

**Reply by RyanFitzSimmonsAK** [reply]
If you remove the filter and scroll through the instance types, is there one labeled `t3.large`, or is it missing entirely?

**Reply by arunppsg** [reply]
`t3.large` is present even when the filter is removed but with unblended cost as 0
```
"Keys": [  "t3.large"   ],
  "Metrics": {  "UnblendedCost": {  "Amount": "0",   "Unit": "USD"  }  }
```

**Comment by RyanFitzSimmonsAK** [comment]
I'm able to get `t3.medium` to show cost in the response.  Could you tell me what version of the CLI you're using?

**Reply by arunppsg** [reply]
I am in 2.9.6

**Reply by RyanFitzSimmonsAK** [reply]
I'd definitely recommend updating to the most recent version, this could be the issue.

**Reply by RyanFitzSimmonsAK** [reply]
Did this work for you?

**Reply by arunppsg** [reply]
Apologies, missed getting back - the issue still persists in the latest version of aws cli (v2.13.26).

**Reply by RyanFitzSimmonsAK** [reply]
Can you check in the AWS Cost Explorer console to see if it looks correct there?

**Reply by arunppsg** [reply]
Yes, it looks correct there.

**Reply by RyanFitzSimmonsAK** [reply]
Could you provide debug logs of this behavior?  You can get debug logs by adding `--debug` to the command, and redacting any sensitive information.  Also, just to verify, do the EC2 instances you created and the IAM user you're using to access the CLI both pertain to the same AWS account that you used to view the Cost Explorer console?
